File: Composicion y Agregacion/ejercicio 3/Python/Parte.py
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso
    def mostrar(self):
        print("Parte:", self.nombre)
        print("Peso:", self.peso, "kg")
class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []
    def agrega(self, parte):
        self.partes.append(parte)
    def mostrar(self):
        print("Avion:", self.modelo, "-", self.fabricante)
        for p in self.partes:
            p.mostrar()
a = Avion("Boeing 737", "Boeing")
a.agrega(Parte("Motor", 2000))
a.agrega(Parte("Alas", 1500))
a.mostrar()

File: Composicion y Agregacion/ejercicio 3/Java/Parte.java
import java.util.ArrayList;
class Partee {
    public String nombre;
    public double peso;
    public Partee(String nombre, double peso) {
        this.nombre = nombre;
        this.peso = peso;
    }
    public void mostrar() {
        System.out.println("Partee: " + nombre + ", Peso: " + peso + " kg");
    }
}
class Avion {
    public String modelo;
    public String fabricante;
    public ArrayList<Partee> partes;
    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }
    public void agregar(Partee parte) {
        partes.add(parte);
    }
    public void mostrar() {
        System.out.println("Avion modelo: " + modelo + ", fabricante: " + fabricante);
        for (Partee p : partes) {
            p.mostrar();
        }
    }
}
public class Parte {
    public static void main(String[] args) {
        Avion a = new Avion("A320", "Airbus");
        a.agregar(new Partee("Motor", 1200));
        a.agregar(new Partee("Alas", 800));
        a.agregar(new Partee("Tren de aterrizaje", 500));
        a.mostrar();
    }
}
